src/bundle/Core/Command/ExpireUserPasswordsCommand.php

```diff
@@ -0,0 +1,383 @@
+                null,
+                InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
+                'Expire password for specific User identified by ID'
+            )
+            ->addOption(
+                'user-group-id',
+                null,
+                InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
+                'Expire passwords of all users assigned to specific User Group'
+            )
+            ->addOption(
+                'user-content-type-identifier',
+                null,
+                InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
+                'Expire passwords of all users based on specific Content Type'
+            )
+            ->addOption(
+                'force',
+                null,
+                InputOption::VALUE_NONE,
+                'Perform setting user passwords as expired'
+            )
+            ->addOption(
+                'batch-size',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'Number of users to process at once',
+                self::DEFAULT_BATCH_SIZE
+            )
+            ->addOption(
+                'password-ttl',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'After how many days passwords expire. Set when Content Type needs to be updated.',
+                self::DEFAULT_PASSWORD_TTL
+            )
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> expires passwords of specific users. 
+Use this tool wisely as next time affect users tries to log in, they will be forced to set a new password.
+Note: This script can potentially run for a very long time, and in Symfony dev environment it will consume memory exponentially with size of dataset.
+
+{$beforeRunningHints}
+EOT
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $batchSize = $input->getOption('batch-size');
+        $userIds = $input->getOption('user-id');
+        $userGroupIds = $input->getOption('user-group-id');
+        $force = $input->getOption('force');
+        $passwordTTL = $input->getOption('password-ttl');
+
+        if (!empty($userIds) && !empty($userGroupIds)) {
+            throw new InvalidArgumentException('You cannot use --user-id and --user-group-id options at once.');
+        }
+
+        $userContentTypeIdentifiers = $input->getOption('user-content-type-identifier');
+
+        $this->summarizeSearchCriteria($output, $userIds, $userGroupIds, $userContentTypeIdentifiers);
+
+        $criteria = $this->getCriteria($userIds, $userGroupIds, $userContentTypeIdentifiers);
+
+        $filter = new Filter(new LogicalAnd($criteria));
+        $filter->withLimit(0);
+
+        $contentList = $this->repository->sudo(static function (Repository $repository) use ($filter): ContentList {
+            return $repository->getContentService()->find($filter);
```

Comment:
`ContentService` should be used directly here

Reply:
WDYM? It needs to be used together with `sudo` as we expect to all users despite permissions.

Reply:
sorry I was thinking about not using a repository here

Reply:
Ok. Fixed: [6687acf](https://github.com/ezsystems/ezplatform-kernel/pull/347/commits/6687acfcdd4e978296fffbf6f1c8240b8a6c5d04)